core/domain/user_query_domain.py

```diff
@@ -24,21 +24,33 @@
 from core import utils
 from core.constants import constants
 
+from typing import List, TypeVar, Type, Optional
+from datetime import datetime
+
+T = TypeVar('T', bound='UserQuery')
+
 attribute_names = [ # pylint: disable=invalid-name
     predicate['backend_attr'] for predicate in (
         constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION)]
 
-UserQueryParams = collections.namedtuple( # pylint: disable=invalid-name
-    'UserQueryParams', attribute_names, defaults=(None,) * len(attribute_names))
+# ignore annotation for namedtuple type
+UserQueryParams = collections.namedtuple( # type: ignore[misc]
+    'UserQueryParams', attribute_names, defaults=(None,) * len(attribute_names)) # type: ignore[misc]
```

Comment:
Why are the ignores needed here?

Reply:
it was based on this response , as mentioned  here the reason for the ignore https://github.com/oppia/oppia/pull/14157#discussion_r740477241

Reply:
This is not done.